web/frontend/src/components/slam.vue

```diff
@@ -169,17 +190,30 @@ const executeMoveOnMap = async () => {
     new grpc.Metadata(),
     (error: ServiceError | null, response: motionApi.MoveOnMapResponse | null) => {
       if (error) {
-        moveClick = !moveClick;
+        allowMove = !allowMove;
+        executeDeleteDestinationMarker();
         toast.error(`Error moving: ${error}`);
         return;
       }
-      moveClick = !moveClick;
+      allowMove = !allowMove;
+      executeDeleteDestinationMarker();
       toast.success(`MoveOnMap success: ${response!.getSuccess()}`);
     }
   );
 
 };
 
+const executeStopMoveOnMap = () => {
```

Comment:
I don't understand what this is doing

Reply:
here we cycle through all active operations.
We can only access this function through the click of the stop button which is only enabled if we've clicked move and the request has not yet been completed.

Reply:
@edaniels let me know that instead of cycling through all active operations, we can use grpc invoke to run the MoveOnMap call, and have it return the op id in the headers even before the MoveOnMap call is finished. See this example: https://github.com/improbable-eng/grpc-web/blob/master/client/grpc-web/docs/invoke.md#example. That way we are definitely canceling the MoveOnMap call that was sent from this browser. I think it's worth spending at least a couple hours to investigate this now, and if it's too challenging, we can file it as an improvement, and maybe ask Fleet to help with a generic utility for doing this.

@micheal-parks does that make sense?

Reply:
@DTCurrie could I get your thoughts on this?

Reply:
That makes sense to me. Only callout is eventually I want to move away from `improbable-eng` over to the `protobuf-es` stuff, but regardless of the runtime we will still probably want some general utility function for setting up cancelable calls and cancelling them. And I am not sure when the ts-sdk team will be able to make that move, because our dependencies on `improbable-eng` run pretty deep.

Reply:
I will create a follow up ticket and mark it as an improvement.

Reply:
I will spend an hour-ish trying to implement this myself.
In the interest of moving forwards, if this is too difficult for me to complete in an hour, I will file a ticket :)